pkg/resource/deploy/step_test.go

```diff
@@ -569,6 +569,242 @@ func TestReadStep(t *testing.T) {
 	})
 }
 
+func TestRefreshStepPatterns(t *testing.T) {
+	t.Parallel()
+
+	// TODO: To some degree, this isn't testing much, since it's mocking the Read and Diff provider calls entirely.
```

Comment:
Same for this TODO, although I'm not sure I agree with what's written here completely.  Hardcoded responses seem fine for engine tests, if anything we should maybe also validate that the engine sends the right parameters, but I don't think we need "real" implementations here.

Reply:
I think we can reword this not to be a TODO because as you say that's kind of the point of these tests -- will sort